src/commonTest/kotlin/arrow/exact/Strings.kt

```diff
@@ -0,0 +1,21 @@
+package arrow.exact
+
+import arrow.core.Either
+import arrow.core.right
+
+class NotBlankString private constructor(val str: String) {
+  companion object : Exact<String, NotBlankString> by exact(String::isNotBlank, ::NotBlankString)
+}
+
+class NotBlankTrimmedString private constructor(val str: String) {
```

Comment:
With my proposed changes this can look like:
```kotlin
class NotBlankTrimmedString private constructor(val str: String) {
  companion object : Exact<String, NotBlankTrimmedString> by exact(::NotBlankTrimmedString) {
     val trimmed = it.trim()
     ensure(trimmed::isNotBlank) { ExactError("blank string") }
     trimmed
  }
}
```

OR

```kotlin
class NotBlankTrimmedString private constructor(val str: String) {
  companion object : Exact<String, NotBlankTrimmedString> by trimmed() and exact(::NotBlankTrimmedString, String::isNotBlank)
}
```